src/auth/passport.js

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable no-underscore-dangle */
+const passport = require('passport');
+const GoogleStrategy = require('passport-google-oauth20').Strategy;
+const GitHubStrategy = require('passport-github').Strategy;
+
+const User = require('../models/user');
+
+passport.use(
+  new GoogleStrategy(
+    {
+      clientID: process.env.GOOGLE_CLIENT_ID,
+      clientSecret: process.env.GOOGLE_CLIENT_SECRET,
+      callbackURL: process.env.GOOGLE_CALL_BACK_URL,
+    },
+    async (accessToken, refreshToken, profile, callBack) => {
+      const user = await User.findOne({ providerId: profile.id });
+
+      if (user) {
+        callBack(null, user);
+      } else {
+        const newUser = new User({
+          email: profile._json.email,
+          name: profile.displayName,
+          providerId: profile.id,
+          provider: profile.provider,
+          password: 'use-your-account-login',
```

Comment:
can't we hash the password and then assign it?

Reply:
when we register/login with third parties they dont offer passwords thats why i assigned to random string because its required in the model, we dont use passwords to login with google/github we just use our accounts